### DISCUSSION TITLE
Duda RadixSort

### DISCUSSION THREAD

**Discussion by mhelmm** [IIC2133-PUC/2024-1#128] [Preguntas materia] [answered]
Hola! En esta diapositiva de clases me surge la duda de cómo se recorren los dígitos en el StableSort llamado por el RadixSort.

![image](https://github.com/IIC2133-PUC/2024-1/assets/105727192/9b981fed-b87c-490f-97a5-d0859bebc265)

En el pseudocódigo el "for" recorre desde 0 hasta d-1, por lo que si llamamos al StableSort se recorrerá de izquierda a derecha. Sin embargo, RadixSort ordena de derecha a izquierda (menos a más significativo). El StableSort se debe encargar de recorrer correctamente a A?

**Comment by Trycoz** [comment]
Me uno a la pregunta. Me surgió la misma duda en esta pregunta vista en la ayudantía 5.

![image](https://github.com/IIC2133-PUC/2024-1/assets/101463191/208c2db6-9f20-455f-825d-1186d4cd22f3)

![image](https://github.com/IIC2133-PUC/2024-1/assets/101463191/0b5b5860-7b4d-4d69-979b-6a94cb675a46)

Se tiene que dentro de RadixSort se hace un "if" para comprobar si es que el dígito se encuentra dentro del rango numérico o alfanumérico. Sin embargo, cuando el "j" es menor a 6 a mi parecer se debe encontrar dentro del rango alfanumérico y no numérico como lo indica la solución. Entiendo que se pueda recorrer al revés pero luego surgen inconsistencias dentro del pseudocódigo (línea 10 y 11) donde se usa el j para acceder a un índice del número entero.

**Reply by darroyue** [reply]
StableSort(A, j) en este caso ordena mirando sólo el j-ésimo digito menos significativo de cada elemento, ignora el resto de los dígitos (tal como lo indica el comentario en el código). Es un CountingSort sólo por ese dígito. Responde eso la duda?

**Comment by Aleksandsk** [comment, marked as answer]
@mhelmm  RadixSort efectivamente ordena de derecha a izquierda, y esto se puede notar cuando dice j-esimo dígito menos significativo, es por esto que cuando j es 0, estas viendo el dígito menos significativo, es decir el de mas a la derecha, y cuando j = d-1, es el d-1 dígito menos significativo, que seria el mas significativo, ósea el de la izquierda, por lo que terminas recorriendo de derecha a izquierda, a su vez StableSort solo se aplica sobre un dígito en especifico de cada palabra de A, por lo que no influye en como se recorren los dígitos en A.

@Trycoz Usando lo anterior, es por esto que los dígitos 0 a 6 representan 0nnnnn, donde el dígito 0 es el n mas a la derecha y dígito 6 el separador 0. Además Counting ordena las palabras por el dígito j , dejando a la palabra con el dígito j mas pequeño a la izquierda y la palabra con el dígito j mas grande a la derecha, por lo que no depende de como se recorran los dígitos de las palabras en RadixSort.
